src/idl/src/annotation.c

```diff
@@ -306,6 +341,85 @@ set_nested(
   return IDL_RETCODE_OK;
 }
 
+static idl_retcode_t
+annotate_default(
+  idl_pstate_t *pstate,
+  idl_annotation_appl_t *annotation_appl,
+  idl_node_t *node)
+{
+  const idl_literal_t *value;
+  idl_type_spec_t *mem_spec;
+
+  assert(annotation_appl);
+  assert(annotation_appl->parameters);
+
+  if (!idl_is_member(node)) {
+    idl_error(pstate, idl_location(annotation_appl),
+      "@default can only be assigned to member nodes");
+    return IDL_RETCODE_SEMANTIC_ERROR;
+  }
+  mem_spec = ((idl_member_t*)node)->type_spec;
+
+  value = annotation_appl->parameters->const_expr;
+  if (!value || !idl_is_literal(value)) {
+    idl_error(pstate, idl_location(annotation_appl),
+      "@default requires a value");
+    return IDL_RETCODE_SEMANTIC_ERROR;
+  }
```

Comment:
This can be an `assert`. The parser checks the syntax and since there's no default, it should throw an error if there's no value specified. Probably good to add a small test for that (if there's none yet). If the test fails, we (or I) should fix the bug there.

Reply:
You're correct that this should not really be checked here.
I will check whether the parser catches this, or if this requires a fix someplace else

Reply:
This is correctly being checked by the parser at a previous step, and therefore I will replace this check with a simpler assert